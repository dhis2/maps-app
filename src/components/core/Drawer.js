import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './styles/Drawer.module.css';

export const Drawer = ({ position = 'right', className, children }) => (
    <div
        className={cx(className, styles.drawer, {
            [styles.left]: position === 'left',
            [styles.right]: position === 'right',
        })}
    >
        {children}
    </div>
);

Drawer.propTypes = {
    position: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]),
};

export default Drawer;
