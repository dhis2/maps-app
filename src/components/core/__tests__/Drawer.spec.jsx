import { render } from '@testing-library/react'
import React from 'react'
import Drawer from '../Drawer.jsx'

describe('Drawer', () => {
    it('should render a right drawer by default', () => {
        const { container } = render(<Drawer />)
        expect(container).toMatchSnapshot()
    })

    it('should render a left drawer if left position', () => {
        const { container } = render(<Drawer position="left" />)
        expect(container).toMatchSnapshot()
    })

    it('should include class name if passed', () => {
        const { container } = render(
            <Drawer position="left" className="myclass" />
        )
        expect(container).toMatchSnapshot()
    })

    it('should render children', () => {
        const { container } = render(
            <Drawer position="left" dataTest="the-data-test-drawer">
                <h4>Title</h4>
                <img src="" />
            </Drawer>
        )

        expect(container).toMatchSnapshot()
    })
})
