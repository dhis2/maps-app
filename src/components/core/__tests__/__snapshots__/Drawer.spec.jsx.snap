// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should include class name if passed 1`] = `
<div>
  <div
    class="myclass drawer left"
  />
</div>
`;

exports[`Drawer should render a left drawer if left position 1`] = `
<div>
  <div
    class="drawer left"
  />
</div>
`;

exports[`Drawer should render a right drawer by default 1`] = `
<div>
  <div
    class="drawer right"
  />
</div>
`;

exports[`Drawer should render children 1`] = `
<div>
  <div
    class="drawer left"
    data-test="the-data-test-drawer"
  >
    <h4>
      Title
    </h4>
    <img
      src=""
    />
  </div>
</div>
`;
